pkg/3scale/amp/operator/base_apimanager_logic_reconciler.go

```diff
@@ -65,31 +66,86 @@ func (r BaseAPIManagerLogicReconciler) ensureOwnerReference(obj common.Kubernete
+	if !r.apiManager.IsPDBEnabled() {
+		common.TagObjectToDelete(desired)
 	}
+	return r.ReconcileResource(&v1beta1.PodDisruptionBudget{}, desired, mutatefn)
+}
+
+func (r *BaseAPIManagerLogicReconciler) ReconcileImagestream(desired *imagev1.ImageStream, mutatefn reconcilers.MutateFn) error {
+	return r.ReconcileResource(&imagev1.ImageStream{}, desired, mutatefn)
+}
+
+func (r *BaseAPIManagerLogicReconciler) ReconcileDeploymentConfig(desired *appsv1.DeploymentConfig, mutatefn reconcilers.MutateFn) error {
+	return r.ReconcileResource(&appsv1.DeploymentConfig{}, desired, mutatefn)
+}
+
+func (r *BaseAPIManagerLogicReconciler) ReconcileService(desired *v1.Service, mutateFn reconcilers.MutateFn) error {
+	return r.ReconcileResource(&v1.Service{}, desired, mutateFn)
+}
+
+func (r *BaseAPIManagerLogicReconciler) ReconcileConfigMap(desired *v1.ConfigMap, mutateFn reconcilers.MutateFn) error {
+	return r.ReconcileResource(&v1.ConfigMap{}, desired, mutateFn)
+}
+
+func (r *BaseAPIManagerLogicReconciler) ReconcileServiceAccount(desired *v1.ServiceAccount, mutateFn reconcilers.MutateFn) error {
+	return r.ReconcileResource(&v1.ServiceAccount{}, desired, mutateFn)
+}
+
+func (r *BaseAPIManagerLogicReconciler) ReconcileRoute(desired *routev1.Route, mutateFn reconcilers.MutateFn) error {
+	return r.ReconcileResource(&routev1.Route{}, desired, mutateFn)
+}
+
+func (r *BaseAPIManagerLogicReconciler) ReconcileSecret(desired *v1.Secret, mutateFn reconcilers.MutateFn) error {
+	return r.ReconcileResource(&v1.Secret{}, desired, mutateFn)
+}
+
+func (r *BaseAPIManagerLogicReconciler) ReconcilePersistentVolumeClaim(desired *v1.PersistentVolumeClaim, mutateFn reconcilers.MutateFn) error {
+	return r.ReconcileResource(&v1.PersistentVolumeClaim{}, desired, mutateFn)
+}
 
-	r.Logger().Info(fmt.Sprintf("Created object %s", ObjectInfo(obj)))
-	return r.Client().Create(context.TODO(), obj) // don't wrap error
+func (r *BaseAPIManagerLogicReconciler) ReconcileRole(desired *rbacv1.Role, mutateFn reconcilers.MutateFn) error {
+	return r.ReconcileResource(&rbacv1.Role{}, desired, mutateFn)
 }
 
-func (r *BaseAPIManagerLogicReconciler) updateResource(obj common.KubernetesObject) error {
-	if err := r.setOwnerReference(obj); err != nil {
+func (r *BaseAPIManagerLogicReconciler) ReconcileRoleBinding(desired *rbacv1.RoleBinding, mutateFn reconcilers.MutateFn) error {
+	return r.ReconcileResource(&rbacv1.RoleBinding{}, desired, mutateFn)
+}
+
+func (r *BaseAPIManagerLogicReconciler) ReconcileResource(obj, desired common.KubernetesObject, mutatefn reconcilers.MutateFn) error {
+	desired.SetNamespace(r.apiManager.GetNamespace())
+	if err := r.setOwnerReference(desired); err != nil {
 		return err
 	}
 
-	r.Logger().Info(fmt.Sprintf("Updated object %s", ObjectInfo(obj)))
-	return r.Client().Update(context.TODO(), obj) // don't wrap error
+	return r.BaseReconciler.ReconcileResource(obj, desired, r.APIManagerMutator(mutatefn))
```

Comment:
If an object processed by APIManager cannot follow what is done by APIManagerMutator, for example, we don't want to add ownerReference to a specific object (because we want to keep it after deletion of the APIManager object itself), what would be the solution for it? Would the idea be to not use the ReconcileResource method and create a custom one for that specific use case?

Reply:
Exactly! 

It is a pity we do not have an example, as all object processed by the apimanager controller receive labels and ensure ownership. 

If there was such an example, it would be using directly  `BaseReconciler.ReconcileResource` method, like this:
```
r.BaseReconciler.ReconcileResource(obj, desired, myMutator)
```

Reply:
:+1: